File: apps/api/src/modules/conversations/conversation.controller.ts
import { Controller, Get, Param, Put, Body, Delete, Post, Patch } from '@nestjs/common';
import { MemoryService } from '../memories/memory.service.js';
import { ConversationService } from './conversation.service.js';

@Controller('conversations')
export class ConversationController {
  constructor(
    private readonly conversationService: ConversationService,
    private readonly memoryService: MemoryService,
  ) {
    this.list = this.list.bind(this);
    this.getById = this.getById.bind(this);
    this.getMemory = this.getMemory.bind(this);
    this.create = this.create.bind(this);
    this.listByAgent = this.listByAgent.bind(this);
    this.patch = this.patch.bind(this);
    this.update = this.update.bind(this);
    this.remove = this.remove.bind(this);
  }

  @Get()
  async list() {
    return this.conversationService.list();
  }

  @Get(':id')
  async getById(@Param('id') id: string) {
    return this.conversationService.getById(id);
  }

  @Get(':id/memory')
  async getMemory(@Param('id') id: string) {
    return this.memoryService.getLatestShortTermMemory(id);
  }

  @Post()
  async create(@Body() body: { agentId: string; title?: string }) {
    return this.conversationService.create({ agentId: body.agentId, title: body.title ?? '新会话' });
  }

  @Get('agent/:agentId')
  async listByAgent(@Param('agentId') agentId: string) {
    return this.conversationService.listByAgent(agentId);
  }

  @Patch(':id')
  async patch(@Param('id') id: string, @Body() body: any) {
    return this.conversationService.update(id, body);
  }

  @Put(':id')
  async update(@Param('id') id: string, @Body() body: any) {
    return this.conversationService.update(id, body);
  }

  @Delete(':id')
  async remove(@Param('id') id: string) {
    return this.conversationService.remove(id);
  }
}

File: apps/server-java/src/main/java/com/haoyitec/agent/server/module/conversation/ConversationController.java
package com.haoyitec.agent.server.module.conversation;

import com.haoyitec.agent.server.module.memory.MemoryService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/conversations")
public class ConversationController {

    private final ConversationService conversationService;
    private final MemoryService memoryService;

    @GetMapping
    public Object list() {
        return conversationService.list();
    }

    @GetMapping("/{id}")
    public Object getById(@PathVariable String id) {
        return conversationService.getById(id);
    }

    @GetMapping("/{id}/memory")
    public Object getMemory(@PathVariable String id) {
        return memoryService.getLatestShortTermMemory(id);
    }

    @PostMapping
    public Object create(@RequestBody Map<String, Object> body) {
        String agentId = body.get("agentId") == null ? null : String.valueOf(body.get("agentId"));
        String title = body.get("title") == null ? "新会话" : String.valueOf(body.get("title"));
        return conversationService.create(agentId, title);
    }

    @GetMapping("/agent/{agentId}")
    public Object listByAgent(@PathVariable String agentId) {
        return conversationService.listByAgent(agentId);
    }

    @PatchMapping("/{id}")
    public Object patch(@PathVariable String id, @RequestBody Map<String, Object> body) {
        return conversationService.update(id, body);
    }

    @PutMapping("/{id}")
    public Object update(@PathVariable String id, @RequestBody Map<String, Object> body) {
        return conversationService.update(id, body);
    }

    @DeleteMapping("/{id}")
    public Object remove(@PathVariable String id) {
        return conversationService.remove(id);
    }
}

